### PR TITLE
refactor(payment): PI-1919 create shared adyen-utils package

### PR DESCRIPTION
**Pull request by vitalii-koshovyi**
## What?
Created shared adyen-utils package for further Adyen GooglePay implementation refactoring

## Why?
Due to the google pay separate package refactoring

## Testing / Proof
Units/manually

image

https://github.com/bigcommerce/checkout-sdk-js/assets/79574476/6f7cdddc-733b-41db-8492-8fe0e3b1ca02

@bigcommerce/team-checkout @bigcommerce/team-payments

### CODE REVIEW

**Review by PavlenkoM — APPROVED** [review]

File: packages/adyen-utils/src/types.ts [hunk 1]
```diff
@@ -943,7 +1028,21 @@ export function isBoletoState(param: unknown): param is BoletoState {
     );
 }
 
-export interface AdyenPaymentMethodInitializationData {
+export function isAccountState(param: any): param is AccountState {
```
Comment (serhii-tkachenko) [inline packages/adyen-utils/src/types.ts:1031]: I have 2 questions around this typeguard:
1. Should we keep it in types file or better move it to a new file?
2. Is it possible to change `any` type to something else?

**Review by serhii-tkachenko — APPROVED** [review]
Nice work 👍

File: packages/adyen-utils/README.md [hunk 2]
```diff
@@ -0,0 +1,11 @@
+# adyen-utils
+
+This library was generated with [Nx](https://nx.dev).
```
Comment (animesh1987) [inline packages/adyen-utils/README.md:3]: Can we add some more descriptive readme here please.

**Review by animesh1987 — APPROVED** [review]
One minor comment, looks good apart from that 👍

**Pull request merged**